src/esp/physics/CollisionGroupHelper.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "CollisionGroupHelper.h"
+
+namespace esp {
+namespace physics {
+
+// initialize the default collision group names
+std::map<std::string, CollisionGroup>
+    CollisionGroupHelper::collisionGroupNames = {
+        {"Default", CollisionGroup::Default},
+        {"Static", CollisionGroup::Static},
+        {"Kinematic", CollisionGroup::Kinematic},
+        {"FreeObject", CollisionGroup::FreeObject},
+        {"Robot", CollisionGroup::Robot},
+        {"Noncollidable", CollisionGroup::Noncollidable},
+        {"UserGroup0", CollisionGroup::UserGroup0},
+        {"UserGroup1", CollisionGroup::UserGroup1},
+        {"UserGroup2", CollisionGroup::UserGroup2},
+        {"UserGroup3", CollisionGroup::UserGroup3},
+        {"UserGroup4", CollisionGroup::UserGroup4},
+        {"UserGroup5", CollisionGroup::UserGroup5},
+        {"UserGroup6", CollisionGroup::UserGroup6},
+        {"UserGroup7", CollisionGroup::UserGroup7},
+        {"UserGroup8", CollisionGroup::UserGroup8},
+        {"UserGroup9", CollisionGroup::UserGroup9},
+};
+
+// initialize the default collision group masks
+std::map<CollisionGroup, CollisionGroups>
+    CollisionGroupHelper::collisionGroupMasks = {
+        // everything except Noncollidable
+        {CollisionGroup::Default,
+         ~CollisionGroups(CollisionGroup::Noncollidable)},
+        // all but Static and Kinematic
+        {CollisionGroup::Static,
+         ~CollisionGroups(CollisionGroup::Static | CollisionGroup::Kinematic |
+                          CollisionGroup::Noncollidable)},
+        // all but Static and Kinematic
+        {CollisionGroup::Kinematic,
+         ~CollisionGroups(CollisionGroup::Static | CollisionGroup::Kinematic |
+                          CollisionGroup::Noncollidable)},
+        // everything except Noncollidable
+        {CollisionGroup::FreeObject,
+         ~CollisionGroups(CollisionGroup::Noncollidable)},
+        // everything except Noncollidable
+        {CollisionGroup::Robot,
+         ~CollisionGroups(CollisionGroup::Noncollidable)},
```

Comment:
Just `~CollisionGroup::Noncollidable` and `~(blah|blah)` elsewhere doesn't work? It should. :thinking:

Reply:
```invalid argument type 'esp::physics::CollisionGroup' to unary expression``` was the error.

Reply:
But that's what `CORRADE_ENUMSET_OPERATORS` should take care of :O

Reply:
Hm, VSCode doesn't like this, but it does compile.